diff --git a/src/asus.c b/src/asus.c
index b8f9671..d7021b7 100644
--- a/src/asus.c
+++ b/src/asus.c
@@ -46,33 +46,66 @@
 #define ASUS_FIELD_RESPONSE	1
 #define ASUS_FIELD_SNAPPING	2
 
-/* key mapping, the index is actual ASUS code */
-static const unsigned char ASUS_KEY_MAPPING[] = {
-/* 00 */	0,		0,		0,		0,
-/* 04 */	KEY_A,		KEY_B,		KEY_C,		KEY_D,
-/* 08 */	KEY_E,		KEY_F,		KEY_G,		KEY_H,
-/* 0C */	KEY_I,		KEY_J,		KEY_K,		KEY_L,
-/* 0E */	KEY_M,		KEY_N,		KEY_O,		KEY_P,
-/* 14 */	KEY_Q,		KEY_R,		KEY_S,		KEY_T,
-/* 18 */	KEY_U,		KEY_V,		KEY_W,		KEY_X,
-/* 1C */	KEY_Y,		KEY_Z,		KEY_1,		KEY_2,
-/* 1E */	KEY_3,		KEY_4,		KEY_5,		KEY_6,
-/* 24 */	KEY_7,		KEY_8,		KEY_9,		KEY_0,
-/* 28 */	KEY_ENTER,	KEY_ESC,	KEY_BACKSPACE,	KEY_TAB,
-/* 2C */	KEY_SPACE,	KEY_MINUS,	KEY_KPPLUS,	0,
-/* 2E */	0,		0,		0,		0,
-/* 34 */	0,		KEY_GRAVE,	KEY_EQUAL,	0,
-/* 38 */	KEY_SLASH,	0,		KEY_F1,		KEY_F2,
-/* 3C */	KEY_F3,		KEY_F4,		KEY_F5,		KEY_F6,
-/* 3E */	KEY_F7,		KEY_F8,		KEY_F9,		KEY_F10,
-/* 44 */	KEY_F11,	KEY_F12,	0,		0,
-/* 48 */	0,		0,		KEY_HOME,	KEY_PAGEUP,
-/* 4C */	KEY_DELETE,	0,		KEY_PAGEDOWN,	KEY_RIGHT,
-/* 4E */	KEY_LEFT,	KEY_DOWN,	KEY_UP,		0,
-/* 54 */	0,		0,		0,		0,
-/* 58 */	0,		KEY_KP1,	KEY_KP2,	KEY_KP3,
-/* 5C */	KEY_KP4,	KEY_KP5,	KEY_KP6,	KEY_KP7,
-/* 5E */	KEY_KP8,	KEY_KP9,	0,
+/* key mapping, the index is actual ASUS code (HID Keyboard/Keypad Usage ID) */
+static const unsigned short ASUS_KEY_MAPPING[] = {
+/* 0x00 */	0,		0,		0,		0,
+/* 0x04 */	KEY_A,		KEY_B,		KEY_C,		KEY_D,
+/* 0x08 */	KEY_E,		KEY_F,		KEY_G,		KEY_H,
+/* 0x0C */	KEY_I,		KEY_J,		KEY_K,		KEY_L,
+/* 0x10 */	KEY_M,		KEY_N,		KEY_O,		KEY_P,
+/* 0x14 */	KEY_Q,		KEY_R,		KEY_S,		KEY_T,
+/* 0x18 */	KEY_U,		KEY_V,		KEY_W,		KEY_X,
+/* 0x1C */	KEY_Y,		KEY_Z,		KEY_1,		KEY_2,
+/* 0x20 */	KEY_3,		KEY_4,		KEY_5,		KEY_6,
+/* 0x24 */	KEY_7,		KEY_8,		KEY_9,		KEY_0,
+/* 0x28 */	KEY_ENTER,	KEY_ESC,	KEY_BACKSPACE,	KEY_TAB,
+/* 0x2C */	KEY_SPACE,	KEY_MINUS,	KEY_EQUAL,	KEY_LEFTBRACE,
+/* 0x30 */	KEY_RIGHTBRACE,	KEY_BACKSLASH,	KEY_BACKSLASH,	KEY_SEMICOLON,
+/* 0x34 */	KEY_APOSTROPHE,	KEY_GRAVE,	KEY_COMMA,	KEY_DOT,
+/* 0x38 */	KEY_SLASH,	KEY_CAPSLOCK,	KEY_F1,		KEY_F2,
+/* 0x3C */	KEY_F3,		KEY_F4,		KEY_F5,		KEY_F6,
+/* 0x40 */	KEY_F7,		KEY_F8,		KEY_F9,		KEY_F10,
+/* 0x44 */	KEY_F11,	KEY_F12,	KEY_SYSRQ,	KEY_SCROLLLOCK,
+/* 0x48 */	KEY_PAUSE,	KEY_INSERT,	KEY_HOME,	KEY_PAGEUP,
+/* 0x4C */	KEY_DELETE,	KEY_END,	KEY_PAGEDOWN,	KEY_RIGHT,
+/* 0x50 */	KEY_LEFT,	KEY_DOWN,	KEY_UP,		KEY_NUMLOCK,
+/* 0x54 */	KEY_KPSLASH,	KEY_KPASTERISK,	KEY_KPMINUS,	KEY_KPPLUS,
+/* 0x58 */	KEY_KPENTER,	KEY_KP1,	KEY_KP2,	KEY_KP3,
+/* 0x5C */	KEY_KP4,	KEY_KP5,	KEY_KP6,	KEY_KP7,
+/* 0x60 */	KEY_KP8,	KEY_KP9,	KEY_KP0,	KEY_KPDOT,
+/* 0x64 */	KEY_102ND,	KEY_COMPOSE,	KEY_POWER,	KEY_KPEQUAL,
+/* 0x68 */	KEY_F13,	KEY_F14,	KEY_F15,	KEY_F16,
+/* 0x6C */	KEY_F17,	KEY_F18,	KEY_F19,	KEY_F20,
+/* 0x70 */	KEY_F21,	KEY_F22,	KEY_F23,	KEY_F24,
+/* 0x74 */	0,		KEY_HELP,	KEY_MENU,	KEY_SELECT,
+/* 0x78 */	KEY_STOP,	KEY_AGAIN,	KEY_UNDO,	KEY_CUT,
+/* 0x7C */	KEY_COPY,	KEY_PASTE,	KEY_FIND,	KEY_MUTE,
+/* 0x80 */	KEY_VOLUMEUP,	KEY_VOLUMEDOWN,	0,		0,
+/* 0x84 */	0,		KEY_KPCOMMA,	KEY_KPEQUAL,	0,
+/* 0x88 */	0,		0,		0,		0,
+/* 0x8C */	0,		0,		0,		0,
+/* 0x90 */	0,		0,		0,		0,
+/* 0x94 */	0,		0,		0,		0,
+/* 0x98 */	0,		KEY_SYSRQ,	KEY_CANCEL,	KEY_CLEAR,
+/* 0x9C */	0,		0,		0,		0,
+/* 0xA0 */	0,		0,		0,		0,
+/* 0xA4 */	0,		0,		0,		0,
+/* 0xA8 */	0,		0,		0,		0,
+/* 0xAC */	0,		0,		0,		0,
+/* 0xB0 */	0,		0,		0,		0,
+/* 0xB4 */	0,		0,		0,		0,
+/* 0xB8 */	0,		0,		0,		0,
+/* 0xBC */	0,		0,		0,		0,
+/* 0xC0 */	0,		0,		0,		0,
+/* 0xC4 */	0,		0,		0,		0,
+/* 0xC8 */	0,		0,		0,		0,
+/* 0xCC */	0,		0,		0,		0,
+/* 0xD0 */	0,		0,		0,		0,
+/* 0xD4 */	0,		0,		0,		0,
+/* 0xD8 */	0,		0,		0,		0,
+/* 0xDC */	0,		0,		0,		0,
+/* 0xE0 */	KEY_LEFTCTRL,	KEY_LEFTSHIFT,	KEY_LEFTALT,	KEY_LEFTMETA,
+/* 0xE4 */	KEY_RIGHTCTRL,	KEY_RIGHTSHIFT,	KEY_RIGHTALT,	KEY_RIGHTMETA,
 };
 
 static const uint8_t ASUS_JOYSTICK_CODES[] = { 0xd0, 0xd1, 0xd2, 0xd3, 0xd7, 0xd8, 0xda, 0xdb };
@@ -127,7 +160,7 @@ asus_code_is_joystick(uint8_t asus_code) {
 
 int
 asus_get_linux_key_code(uint8_t asus_code) {
-	if (asus_code > ARRAY_LENGTH(ASUS_KEY_MAPPING)) {
+	if (asus_code >= ARRAY_LENGTH(ASUS_KEY_MAPPING)) {
 		return -1;
 	}
 	return ASUS_KEY_MAPPING[asus_code];
