diff -ur a/ b/
--- a/src/preload.js      2021-09-27 18:51:11.900797800 -0700
+++ b/src/preload.js      2021-12-17 10:25:44.803464279 -0800
@@ -8,3 +8,3 @@
 const { webFrame } = require('electron');
-global.NEW_BACKEND = !process.versions.electron.startsWith('13') && process.contextIsolated;
+global.NEW_BACKEND = true;
 
 require('../polyfills');
--- a/src/browserWindow.js      2021-09-27 18:51:11.900797800 -0700
+++ b/src/browserWindow.js      2021-12-17 10:25:44.803464279 -0800
@@ -21,3 +21,4 @@
   constructor (opts) {
     let originalPreload;
+    if (!opts.webPreferences) opts.webPreferences = {};
     if (opts.webContents) {
--- a/src/browserWindow.js      2021-09-27 18:51:11.900797800 -0700
+++ b/src/browserWindow.js      2021-12-17 10:25:44.803464279 -0800
@@ -52,5 +52,6 @@
     }
 
+    opts.webPreferences.contextIsolation = true;
     const win = new BrowserWindow(opts);
     const ogLoadUrl = win.loadURL.bind(win);
