From 5c90dd84f541bd4789525f12f12ad24411b99018 Mon Sep 17 00:00:00 2001
From: Thomas Tuegel <ttuegel@mailbox.org>
Date: Mon, 27 Jan 2020 05:31:23 -0600
Subject: [PATCH 2/3] xwayland

---
 src/xwl/xwayland.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xwl/xwayland.cpp b/src/xwl/xwayland.cpp
index 57efdde..a211a58 100644
--- a/xwl/xwayland.cpp
+++ b/xwl/xwayland.cpp
@@ -124,5 +124,5 @@ void Xwayland::start()
 
     xwayland_process->setProcessChannelMode(QProcess::ForwardedErrorChannel);
-    xwayland_process->setProgram(QStringLiteral("Xwayland"));
+    xwaylandProcess->setProgram(QLatin1String(NIXPKGS_XWAYLAND));
 
     QProcessEnvironment env = app->processStartupEnvironment();
-- 
2.29.2

